src/group_theory/submonoid/default.lean

```diff
@@ -0,0 +1,387 @@
+/-
+Copyright (c) 2018 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Kenny Lau, Johan Commelin, Mario Carneiro, Kevin Buzzard,
+Amelia Livingston, Yury Kudryashov
+-/
+import group_theory.submonoid.basic
+import algebra.big_operators
+import algebra.free_monoid
+import algebra.group.prod
+import data.equiv.mul_add
+
+/-!
+# Submonoids
```

Comment:
`submonoid.default` and `submonoid.basic` should have different titles in the docs and the docs should also explain the criterion for inclusion in one file vs the other.

Reply:
Currently I wanted a small file with few dependencies (e.g., not depending on `finset` or `nat.basic`) that leads to `closure`. I guess for consistency I should either move `map`/`comap` to `default` or move definitions/lemmas not relying on extra `import`s to `basic`.

Reply:
Maybe `default` should renamed something like `operations` and a new `default` file which just imports `basic` and `operations` should be added?

Reply:
I like that suggestion